lib/webidl2.js

```diff
@@ -823,6 +826,13 @@
       all_ws(isPartial ? null : store, "pea");
       if (!consume(ID, "dictionary")) return;
       all_ws();
+      const notAllowed = Array.isArray(store) && store
```

Comment:
`store` is only defined with `ws` option (`parse(text, { ws: true })`), and its purpose is to collect whitespaces. It's totally not being tested and is undocumented.

Reply:
heh, whoops :) Ok, but it's the thing that contains the extended attribute objects? Would we need to move those somehow?

Reply:
No, it contains previous top level definitions including interfaces, enums, etc., so that any whitespaces between them can be mixed in the list.

Reply:
I think we should explicitly pass extattrs list to `dictionary()`.

https://github.com/w3c/webidl2.js/blob/96943e30d486ed949cbfe0e8315da92f9b114910/lib/webidl2.js#L967

This way `definition()` should receive it first and pass it to dictionary parser.

Reply:
Yeah, that would be nice actually. I noticed it called the function twice.

Reply:
BTW, shouldn't it be *error if anything else but an interface has NoInterfaceObject*?

Reply:
Yeah, I was thinking the same. Might need to rewrite all this to clarify things for each type ("interfaces shouldn't have of these: x, y, z", "dictionaries shouldn't have any of these: w,z", and so on).